gcloud/monitoring/query.py

```diff
@@ -0,0 +1,661 @@
+
+            resource.type = "<value>"
+
+        See the `defined resource types`_.
+
+        .. note::
+
+            The label ``"instance_name"`` is a metric label,
+            not a resource label. You would filter on it using
+            ``select_metrics(instance_name=...)``.
+
+        :param args: Raw filter expression strings to include in the
+            conjunction. If just one is provided and no keyword arguments
+            are provided, it can be a disjunction.
+
+        :param kwargs: Label filters to include in the conjunction as
+            described above.
+
+        :rtype: :class:`Query`
+        :returns: The new query object.
+
+        .. _defined resource types:
+            https://cloud.google.com/monitoring/api/v3/monitored-resources
+        """
+        new_query = self.copy()
+        new_query._filter.select_resources(*args, **kwargs)
+        return new_query
+
+    def select_metrics(self, *args, **kwargs):
+        """Copy the query and add filtering by metric labels.
+
+        Examples::
+
+            query = query.select_metrics(instance_name='myinstance')
+            query = query.select_metrics(instance_name_prefix='mycluster-')
+
+        A keyword argument ``<label>=<value>`` ordinarily generates a filter
+        expression of the form::
+
+            metric.label.<label> = "<value>"
+
+        However, by adding ``"_prefix"`` or ``"_suffix"`` to the keyword,
+        you can specify a partial match.
+
+        ``<label>_prefix=<value>`` generates::
+
+            metric.label.<label> = starts_with("<value>")
+
+        ``<label>_suffix=<value>`` generates::
+
+            metric.label.<label> = ends_with("<value>")
+
+        :param args: Raw filter expression strings to include in the
+            conjunction. If just one is provided and no keyword arguments
+            are provided, it can be a disjunction.
+
+        :param kwargs: Label filters to include in the conjunction as
+            described above.
+
+        :rtype: :class:`Query`
+        :returns: The new query object.
+        """
+        new_query = self.copy()
+        new_query._filter.select_metrics(*args, **kwargs)
+        return new_query
+
+    def align(self, per_series_aligner, seconds=0, minutes=0, hours=0):
+        """Copy the query and add temporal alignment.
+
+        If ``per_series_aligner`` is not ``Aligner.ALIGN_NONE``, each time
+        series will contain data points only on the period boundaries.
+
+        Example::
+
+            query = query.align(Aligner.ALIGN_MEAN, minutes=5)
+
+        It is also possible to specify the aligner as a literal string::
+
+            query = query.align('ALIGN_MEAN', minutes=5)
+
+        :type per_series_aligner: string
+        :param per_series_aligner: The approach to be used to align
+            individual time series. For example: ``Aligner.ALIGN_MEAN``.
+            See :class:`Aligner` and the descriptions of the `supported
+            aligners`_.
```

Comment:
I'm not sure the link will actually work when broken across lines. Have you verified?